src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -83,10 +83,18 @@ enum ResolutionFailure<'a> {
     /// this could be an enum variant, but the last path fragment wasn't resolved.
     /// the `String` is the variant that didn't exist
     NotAVariant(Res, Symbol),
+    /// This link has malformed generic parameters; i.e., the angle brackets are mismatched.
+    /// For example, `Vec<T` should trigger this, as should `Vec<T>>`.
+    MalformedGenerics(MalformedGenericsVariant),
```

Comment:
Do you expect to ever add more variants? If not this seems a little overkill.

Reply:
I added an enum for a few reasons:

- In the future we might want to check that generics are well-formed wrt the type system; e.g. not allowing `Vec<>`
- In the future we may implement more advanced generic links where rustdoc will link to the generic parameter if it is a concrete type; e.g. linking to `Vec` _and_ `u32` in `Vec<u32>`
- I wanted to group anything related to an issue with generics together instead of having a variant of `ResolutionFailure` named something like `UnbalancedAngleBracketsInGenerics`

Reply:
I think we should wait to generalize until we need to generalize. When those things are implemented, then we can add an enum.

Reply:
Okay, so should I change the variant to `UnbalancedAngleBracketsInGenerics` then?

Reply:
I would prefer it to just be called `MalformedGenerics` for now.

Reply:
The reason I prefer `UnbalancedAngleBrackets...` is because the error message says that there are unbalanced angle brackets; if the variant's name is more general then people may try to use it and the error message will be unexpected. I would prefer a name that's less surprising even if it's longer.

Reply:
Oh, I see what you're saying - you want it to be clear what to do for the next person who comes along just by reading the code.

I guess in that case it makes sense to have the enum even if it is a little overkill.